packages/ses/src/module-proxy.js

```diff
@@ -0,0 +1,173 @@
+      if (!isExtensible(module)) {
+        preventExtensions(target);
+      }
+      if (isSealed(module)) {
+        seal(target);
+      }
+    },
+    module: new Proxy(target, {
+      get(_target, name, receiver) {
+        if (internalModule === undefined) {
+          throw new TypeError(
+            `Cannot get property ${q(
+              name,
+            )} of module exports namespace, the module has not yet begun executing`,
+          );
+        }
+        return Reflect.get(internalModule, name, receiver);
+      },
+      set(_target, name, value) {
+        if (internalModule === undefined) {
+          throw new TypeError(
+            `Cannot set property ${q(
+              name,
+            )} of module exports namespace, the module has not yet begun executing`,
+          );
+        }
+        if (!Reflect.set(internalModule, name, value)) {
+          throw new TypeError(
+            `Cannot add property ${q(
+              name,
+            )}, the module exports namespace is not extensible`,
+          );
+        }
+        return true;
+      },
+      has(_target, name) {
+        if (internalModule === undefined) {
+          throw new TypeError(
+            `Cannot check property ${q(
+              name,
+            )}, the module has not yet begun to execute`,
+          );
+        }
+        return Reflect.has(target, name);
+      },
+      deleteProperty(_target, name) {
+        throw new TypeError(
+          `Cannot delete property ${q(
+            name,
+          )}, the module exports namespace is not extensible`,
+        );
+      },
+      ownKeys(_target) {
+        if (internalModule === undefined) {
+          throw new TypeError(
+            `Cannot enumerate keys, the module has not yet begun to execute`,
+          );
+        }
+        return Reflect.ownKeys(internalModule);
+      },
+      getOwnPropertyDescriptor(_target, name) {
+        if (internalModule === undefined) {
+          throw new TypeError(
+            `Cannot get own property descriptor ${q(
+              name,
+            )}, the module has not yet begun to execute`,
+          );
+        }
+        return Reflect.getOwnPropertyDescriptor(internalModule, name);
+      },
+      preventExtensions(_target) {
+        if (internalModule === undefined) {
+          throw new TypeError(
+            `Cannot prevent extensions to module exports namespace until the module has not yet begun to execute`,
+          );
+        }
+        preventExtensions(target);
+        preventExtensions(internalModule);
+        return true;
+      },
+      isExtensible() {
+        if (internalModule === undefined) {
+          return Reflect.isExtensible(target);
+        }
+        return Reflect.isExtensible(internalModule);
+      },
+      getPrototypeOf(_target) {
+        return null;
+      },
+      setPrototypeOf(_target, _proto) {
+        throw new TypeError(`Cannot set prototype of module exports namespace`);
```

Comment:
To the user, the proxy is the "module namespace object", whereas the `internalModule` is the "module exports namespace", yes? If so, the user who sees this error would have been trying to set the prototype of the module namespace object (or just "module namespace")

Reply:
Exactly. That the module exports namespace they see is a proxy should not be mentioned in an error message.

Reply:
Since this feedback, I’ve made these names more clear: “exports proxy” and “proxied exports”. The “proxied exports” is the target of the “exports proxy”, so there isn’t a third name.